x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/date/DateExtract.java

```diff
@@ -63,11 +62,7 @@ public ExpressionEvaluator.Factory toEvaluator(Function<Expression, ExpressionEv
         var fieldEvaluator = toEvaluator.apply(children().get(1));
         if (children().get(0).foldable()) {
             ChronoField chrono = chronoField();
-            if (chrono == null) {
-                BytesRef field = (BytesRef) children().get(0).fold();
-                throw new InvalidArgumentException("invalid date field for [{}]: {}", sourceText(), field.utf8ToString());
```

Comment:
I agree that we shouldn't throw an IAE - as that becomes a 500 response.

I would prefer if we'd return a 400 response by throwing a `VerificationException`, though, like for date math [overflows during folding](https://github.com/elastic/elasticsearch/blob/d95d6adae5fd76d2e7cef620cf963cee55d85fb2/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/predicate/operator/arithmetic/DateTimeArithmeticOperation.java#L127) - rather than returning null + a warning.

The last status in https://github.com/elastic/elasticsearch/issues/100038 is that we still have to decide whether we want to consistently return null+warning or a 400 response; I think we should take that decision before merging this :)

Reply:
We need to extract validation of arguments in a dedicated method as it occurs _after_ typeResolution and _before_ execution. See #105425

Reply:
> I agree that we shouldn't throw an IAE - as that becomes a 500 response.
> 
> I would prefer if we'd return a 400 response by throwing a `VerificationException`, though, like for date math [overflows during folding](https://github.com/elastic/elasticsearch/blob/d95d6adae5fd76d2e7cef620cf963cee55d85fb2/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/predicate/operator/arithmetic/DateTimeArithmeticOperation.java#L127) - rather than returning null + a warning.
> 
> The last status in #100038 is that we still have to decide whether we want to consistently return null+warning or a 400 response; I think we should take that decision before merging this :)

I was debating with myself whether we should error out or return null with a warning for such cases. In the CsvTest, there are a few date related test cases, when there is an invalid date format provided as inputs, we return null with warnings, I'm kind of leaning toward a warning here. One reason is that there are cases we may not want this kind of invalid format fail the entire query or block the whole workload, it might be one of many returned field/column in the resultset. And second is that it is consistent with when it comes from an index field, not a literal value.

Reply:
> We need to extract validation of arguments in a dedicated method as it occurs _after_ typeResolution and _before_ execution. See #105425

If we want to make the behavior consistent between the cases - when the arguments are provided as literal values vs the arguments come from index fields(known at execution time), I guess we may need to validate it at the very beginning of the execution.